pkg/tfgen/generate_schema.go

```diff
@@ -553,6 +554,48 @@ func (g *schemaGenerator) genProperty(prop *variable) pschema.PropertySpec {
 			if i, ok := defaultValue.(int); ok {
 				defaultValue = float64(i)
 			}
+			rt := reflect.TypeOf(defaultValue)
+			if rt != nil {
+				switch kind := rt.Kind(); kind {
+				case
+					reflect.Bool,
+					reflect.Int,
+					reflect.Int8,
+					reflect.Int16,
+					reflect.Int32,
+					reflect.Int64,
+					reflect.Uint,
+					reflect.Uint8,
+					reflect.Uint16,
+					reflect.Uint32,
+					reflect.Uint64,
+					reflect.Float32,
+					reflect.Float64,
+					reflect.String:
+					// These are fine.
+				case
+					reflect.Uintptr,
+					reflect.Complex64,
+					reflect.Complex128,
+					reflect.Array,
+					reflect.Chan,
+					reflect.Func,
+					reflect.Interface,
+					reflect.Map,
+					reflect.Pointer,
+					reflect.Slice,
+					reflect.Struct,
+					reflect.UnsafePointer:
+					fallthrough
```

Comment:
We don't need to list all of these.

Reply:
I added this after a discussion with @t0yv0. I might have misunderstood what he meant but I believe the argument was that having all cases here makes why we are doing this more explicit to a reader.